core/src/main/java/com/oath/oak/InternalOakMap.java

```diff
@@ -141,6 +141,24 @@ int entries() {
         return size.get();
     }
 
+    /*-------------- Context --------------*/
+
+    // This array holds for each thread a context instance to avoid redundant instantiation and GC.
+    private final ThreadContext[] threadLocalContext = new ThreadContext[ThreadIndexCalculator.MAX_THREADS];
+    private final ThreadIndexCalculator contextThreadIndexCalculator = ThreadIndexCalculator.newInstance();
+
+    /**
+     * Should only be called from API methods at the beginning of the method and be reused in internal calls.
+     * @return a thread-local context instance.
+     */
+    ThreadContext getThreadLocalContext() {
```

Comment:
Is a context ever fully reset/invalidated? Why can't it be that some old settings will interfere erroneously?

Reply:
Should be invalidated when we get the context here.